ocaml-lsp-server/test/e2e-new/type_enclosing.ml

```diff
@@ -0,0 +1,431 @@
+open Test.Import
+
+let call_type_enclosing ?(verbosity = 0) ?range_end client position index =
+  let uri = DocumentUri.of_path "test.ml" in
+  let text_document = TextDocumentIdentifier.create ~uri in
+  let params =
+    `Assoc
+      ([ ("textDocument", TextDocumentIdentifier.yojson_of_t text_document)
+       ; ("position", Position.yojson_of_t position)
+       ; ("index", `Int index)
+       ; ("verbosity", `Int verbosity)
+       ]
+      @
+      match range_end with
+      | None -> []
+      | Some x -> [ ("rangeEnd", Position.yojson_of_t x) ])
+  in
+  let params = Some (Jsonrpc.Structured.t_of_yojson params) in
+  let req =
+    Lsp.Client_request.UnknownRequest
+      { meth = "ocamllsp/typeEnclosing"; params }
+  in
+  Client.request client req
+
+let print_type_enclosing result =
+  result |> Yojson.Safe.pretty_to_string ~std:false |> print_endline
+
+let%expect_test "type enclosing on simple example" =
+  let source = {|let x = string_of_int 2002|} in
+  let request client =
+    let open Fiber.O in
+    let position = Position.create ~line:0 ~character:22 in
+    let index = 0 in
+    let+ response = call_type_enclosing client position index in
+    print_type_enclosing response
+  in
+  Helpers.test source request;
+  [%expect
+    {|
+    {
+      "index": 0,
+      "enclosings": [
+        {
+          "end": { "character": 26, "line": 0 },
+          "start": { "character": 8, "line": 0 }
+        }
+      ],
+      "type": "int"
```

Comment:
This is wrong isn't it ? 

Looks like the enclosing of `string_of_int 2002` which should have type string...
We expect something like this: 

```    {
      "index": 0,
      "enclosings": [
        {
          "end": { "character": 26, "line": 0 },
          "start": { "character": 22, "line": 0 }
        },
        {
          "end": { "character": 26, "line": 0 },
          "start": { "character": 8, "line": 0 }
        }
      ],
      "type": "int"
}
```

And if the required index was 1:
```    {
      "index": 1,
      "enclosings": [
        {
          "end": { "character": 26, "line": 0 },
          "start": { "character": 22, "line": 0 }
        },
        {
          "end": { "character": 26, "line": 0 },
          "start": { "character": 8, "line": 0 }
        }
      ],
      "type": "string"
}
```